### PR TITLE
Run actionlint static checker pre-merge.

### PR DESCRIPTION
**Pull request by sengi**
https://github.com/rhysd/actionlint

We badly need this to help prevent more easily-avoidable vulns and bugs creeping into the reusable actions/workflows. Probably should have added this a long time ago, but hindsight etc. etc.

Fix all the pre-existing lints so that we can make this a mandatory check.

Tested: successfully [ran the multi-arch build workflow](https://github.com/alphagov/release/actions/runs/9140877804) and the [Brakeman workflow](https://github.com/alphagov/release/actions/runs/9131497859/job/25110816677). The others aren't so easy to test but they're also not that operationally important so we can just fix forward if I've introduced any issues.

### CODE REVIEW

**Comment by sengi** [comment]
Heh, this might take a while. Almost wish I hadn't looked underneath this rock 😂

File: .github/actionlint-matcher.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "problemMatcher": [
+    {
+      "owner": "actionlint",
+      "pattern": [
+        {
+          "regexp": "^(?:\\x1b\\[\\d+m)?(.+?)(?:\\x1b\\[\\d+m)*:(?:\\x1b\\[\\d+m)*(\\d+)(?:\\x1b\\[\\d+m)*:(?:\\x1b\\[\\d+m)*(\\d+)(?:\\x1b\\[\\d+m)*: (?:\\x1b\\[\\d+m)*(.+?)(?:\\x1b\\[\\d+m)* \\[(.+?)\\]$",
```
Comment (sengi) [inline .github/actionlint-matcher.json:7]: I don't like that this is matching based on the colours, but meh, it works and I don't want to rewrite it myself :/

**Comment by sengi** [comment]
RFAL

**Review by dj-maisy — APPROVED** [review]
🥇

**Comment by nimalank7** [comment]
Closes https://github.com/alphagov/govuk-infrastructure/pull/1312

**Pull request merged**